pbj-integration-tests/src/main/java/com/hedera/pbj/integration/fuzz/SingleFuzzTest.java

```diff
@@ -0,0 +1,109 @@
+package com.hedera.pbj.integration.fuzz;
+
+import com.hedera.pbj.runtime.Codec;
+import com.hedera.pbj.runtime.io.buffer.BufferedData;
+
+import java.util.Random;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * A single fuzz test.
+ * <p>
+ * This class exposes a single public static method that allows a caller to pass
+ * a valid object, a Protobuf codec for this object, as well as a random numbers
+ * generator. The method will then perform a single fuzz test run with the given
+ * data and return a SingleFuzzTestResult describing the outcome of the test run.
+ * <p>
+ * A fuzz testing framework is expected to use this class to execute multiple runs
+ * of the same test (each being random per the given random numbers generator)
+ * until the set of outcomes satisfies the testing framework threshold,
+ * or the framework runs out of the maximum number of attempts, or a timeout
+ * occurs.
+ * <p>
+ * This class is thread-safe and can be called concurrently from multiple threads
+ * as long as the input parameters are immutable or thread-safe.
+ */
+public final class SingleFuzzTest {
+    private final static AtomicInteger TEST_ID_GENERATOR = new AtomicInteger(0);
+
+    private static <T> BufferedData write(final T object, final Codec<T> codec, final int size) throws Exception {
```

Comment:
As far as I remember, `Codec` has a method to measure size of object, in bytes. So `size` here is probably redundant.

Reply:
You're correct. However, the code calling this method has already called codec.measureRecord() at lines 53 and 84 below. The implementation of the measureRecord() is such that it iterates over all the fields of an object, which may be rather expensive for large objects. Since I'll need to run this test many times in order to gather statistics (because a single fuzz test run may not reveal issues), I figured I'd optimize it a bit here and not call measureRecord() the second time since I already have the value. Please let me know what you think.